daemon/containerd/image.go

```diff
@@ -158,15 +163,17 @@ func (i *ImageService) GetImageManifest(ctx context.Context, refOrID string, opt
 	}
 
 	if containerdimages.IsManifestType(desc.MediaType) {
+		if options.Platform != nil && desc.Platform == nil {
+			return nil, errdefs.NotFound(errors.New("mismatched platform"))
+		}
+		if options.Platform != nil && desc.Platform != nil && !platform.Match(*options.Platform) {
+			return nil, errdefs.NotFound(errors.New("mismatched platform"))
```

Comment:
Wondering if we need to make the error messages more informative (so: "what's the platform of the image then, if it doesn't match?")

There's some other discussions around that, so perhaps it's ok to do so in a follow-up, but perhaps a "todo" could make sense